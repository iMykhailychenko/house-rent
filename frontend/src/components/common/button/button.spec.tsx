import { fireEvent, render } from '@testing-library/react';

import Button from './button';

describe('Button', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<Button>test button</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot props', () => {
        const { asFragment, queryByAltText } = render(
            <Button primary type="submit" className="test">
                test button
            </Button>,
        );
        expect(asFragment()).toMatchSnapshot();
        expect(queryByAltText(/loading/i)).toBeNull();
    });

    it('click event', () => {
        const handleClick = jest.fn();
        const screen = render(<Button onClick={handleClick}>test button</Button>);
        fireEvent.click(screen.getByText(/test button/i).parentNode as Element);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('loading', () => {
        const screen = render(<Button loading>test button</Button>);
        expect(screen.getByAltText(/loading/i)).toBeInTheDocument();
        expect(screen.getByAltText(/loading/i)).toHaveAttribute('src');
    });
});
