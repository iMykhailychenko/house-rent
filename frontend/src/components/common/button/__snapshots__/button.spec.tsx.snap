// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    <span
      class="children"
    >
      test button
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button matches snapshot props 1`] = `
<DocumentFragment>
  <button
    class="button test primary"
    type="submit"
  >
    <span
      class="children"
    >
      test button
    </span>
  </button>
</DocumentFragment>
`;
